From 8bf168172798132cd08989ba6a756a430856d38e Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Thu, 17 Sep 2026 10:22:25 +0200
Subject: [PATCH] broadway: add missing generated header deps to gtk4-broadwayd

Fixes race condition where gdkversionmacros.h doesnt exist

	[473/1702] Compiling C object gdk/broadway/gtk4-broadwayd.p/broadway-server.c.o
	ninja: job failed with status 1: cc -Igdk/broadway/gtk4-broadwayd.p -Igdk/broadway -I../gdk/broadway -I. -I.. -Igdk -I../gdk -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/glycin-2 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/fribidi -I/usr/include/graphene-1.0 -I/usr/lib/graphene-1.0/include -I/usr/include/libdrm -I/usr/include/gio-unix-2.0 -I/usr/include/webp -I/usr/include/gstreamer-1.0 -I/usr/include/orc-0.4 -I/usr/lib/gstreamer-1.0/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu11 -DG_LOG_USE_STRUCTURED=1 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_90 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_90 '-DPANGO_VERSION_MIN_REQUIRED=G_ENCODE_VERSION(1,58)' '-DPANGO_VERSION_MAX_ALLOWED=G_ENCODE_VERSION(1,59)' '-DGTK_VERSION="4.24.0"' -D_GNU_SOURCE -Os -fstack-clash-protection -Wformat -Werror=format-security -fno-plt -g -O2 -O2 -fPIE -pthread -mfpmath=sse -msse -msse2 -DGTK_COMPILATION '-DG_LOG_DOMAIN="Gdk"' -MD -MQ gdk/broadway/gtk4-broadwayd.p/broadway-server.c.o -MF gdk/broadway/gtk4-broadwayd.p/broadway-server.c.o.d -o gdk/broadway/gtk4-broadwayd.p/broadway-server.c.o -c ../gdk/broadway/broadway-server.c
	In file included from ../gdk/gdktypes.h:45,
	                 from ../gdk/broadway/broadway-server.c:9:
	../gdk/gdkenums.h:33:10: fatal error: gdk/version/gdkversionmacros.h: No such file or directory
	   33 | #include <gdk/version/gdkversionmacros.h>
	      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
	compilation terminated.
	ninja: subcommand failed

Part-of: <https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/10383>
---
 gdk/broadway/meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gdk/broadway/meson.build b/gdk/broadway/meson.build
index 1579c1fda53..fb3e8b75ddd 100644
--- a/gdk/broadway/meson.build
+++ b/gdk/broadway/meson.build
@@ -72,6 +72,8 @@ executable('gtk4-broadwayd',
     broadwayjs_h,
     gdkconfig,
     gdkenum_h,
+    gdkversionmacros_h,
+    gdk_visibility_h,
     'broadwayd.c',
     'broadway-server.c',
     'broadway-output.c',
-- 
GitLab

